buildrunner/steprunner/tasks/run.py

```diff
@@ -387,14 +387,16 @@ def _archive_file(
                 f"Error gathering artifact {artifact_file}",
             )
 
-        # register the artifact with the run controller
-        self.step_runner.build_runner.add_artifact(
-            os.path.join(
-                self.step_runner.name,
-                output_file_name,
-            ),
-            properties,
-        )
+        if properties is None or (isinstance(properties, OrderedDict) \
+            and ('push' not in properties.keys() or ('push' in properties.keys() and properties['push'] is True))):
```

Comment:
This check can be simplified to `if not properties or (isinstance(properties, dict) and properties.get('push', True))`

Reply:
Nice optimization! I put that in.